lib/Cache/RedisDB.pm

```diff
@@ -75,26 +78,56 @@ sub redis {
     return $redis;
 }
 
-=head2 get($namespace, $key)
+=head2 get
+
+Takes a C<$namespace> and C<$key> parameter, and returns the scalar value
+corresponding to that cache entry.
 
-Retrieve I<$key> value from the cache.
+This will automatically deserialise data stored with L<Sereal>. If no data
+is found, this will return C<undef>.
 
 =cut
 
 sub get {
     my ($self, $namespace, $key) = @_;
-    my $res = redis->get(_cache_key($namespace, $key));
+    my $res = redis()->get(_cache_key($namespace, $key));
     if (looks_like_sereal($res)) {
         state $decoder = Sereal::Decoder->new();
         $res = $decoder->decode($res);
     }
     return $res;
 }
 
-=head2 set($namespace, $key, $value[, $exptime])
+=head2 mget
+
+Retrieve values for multiple keys in a single call.
 
-Assigns I<$value> to the I<$key>. I<$value> should be scalar value.
-If I<$exptime> specified, it is expiration time in seconds.
+Similar to L</get>, this takes a C<$namespace> as the first parameter,
+but it also accepts a list of C<@keys> to look up.
+
+Returns an arrayref in the same order as the original keys. For any
+key that had no value, the resulting arrayref will contain C<undef>.
+
+=cut
+
+sub mget {
+    my ($self, $namespace, @keys) = @_;
+    my $res = Cache::RedisDB::redis()->mget(map { Cache::RedisDB::_cache_key($namespace, $_) } @keys);
+    state $decoder = Sereal::Decoder->new();
```

Comment:
I see the similar line in `get`. May be unite them and use single decoder instance?